docs/source/user_guide/basic.rst

```diff
@@ -180,34 +181,39 @@ We can also use :meth:`cupy.ndarray.get()`:
 Memory management
 -----------------
 
-Check :doc:`./memory` for a detailed description of how is memory managed in CuPy
+Check :doc:`./memory` for a detailed description of how memory is managed in CuPy
 using memory pools.
 
 
-How to write CPU/GPU agnostic code
+How to write device agnostic code
 ----------------------------------
 
-The compatibility of CuPy with NumPy enables us to write CPU/GPU generic code.
-It can be made easy by the :func:`cupy.get_array_module` function.
-This function returns the :mod:`numpy` or :mod:`cupy` module based on arguments.
-A CPU/GPU generic function is defined using it like follows:
+CuPy's compatibility with NumPy makes it possible to write device agnostic code.
+For this purpose, CuPy implements the :func:`cupy.get_array_module` function, that
+returns a reference to :mod:`cupy` if any of its arguments resides on a GPU
+and :mod:`numpy` otherwise.
+Here is an example of a device agnostic function that computes ``log1p``:
 
 .. doctest::
 
    >>> # Stable implementation of log(1 + exp(x))
    >>> def softplus(x):
-   ...     xp = cp.get_array_module(x)
-   ...     return xp.maximum(0, x) + xp.log1p(xp.exp(-abs(x)))
+   ...     mod = cp.get_array_module(x)
+   ...     print("Using module:", mod.__name__)
+   ...     return mod.maximum(0, x) + mod.log1p(mod.exp(-abs(x)))
 
-Sometimes, an explicit conversion to a host or device array may be required.
-:func:`cupy.asarray` and :func:`cupy.asnumpy` can be used in agnostic implementations
-to get host or device arrays from either CuPy or NumPy arrays.
+When you need to operate on arrays that reside on CPU and GPU,
+you first need to transfer them to the same device -- either CPU or GPU.
```

Comment:
The new description reads odd to me, mainly because 1. I don't see it offers better clarity compared to the old one, and 2. "CPU" usually is not considered a device. How about this?
```suggestion
When you need to manipulate CPU and GPU arrays, an explicit data
transfer may be required to move them to the same location -- either CPU or GPU.
```